momentum/4/lua/ref-msys-tls-params-set.md

```diff
@@ -0,0 +1,54 @@
+---
+title: "msys.tls_params.set"
+description: "msys tls_params set Set a tls parameter string in the Momentum configuration, overriding if the parameter is already set"
+---
+
+<a name="lua.ref.msys.tls_params.set"></a> 
+## Name
+
+msys.tls_params.set — Set a tls parameter string in the Momentum configuration, overriding if the parameter is already set.
+
+## Synopsis
+
+`require('msys.tls_params');`
+
+`msys.tls_params.set(tls_params, "param", "val");`
+
+```
+tls_params: mixed
+param: string
+val: string
+```
+
+## Description
+
+This function sets a tls parameter string in the Momentum configuration, overriding if the parameter is already set.
+
+<a name="lua.ref.msys.tls_params.set.example"></a> 
+
+The following example shows how to use this function in the `get_outbound_tls_parameters` hook to override all outbound tls parameters.
+It is important to note that the use of this hook will reset outbound tls config for the current connection, requiring the user to set a value for all keys listed in the example below.
```

Comment:
All keys below are required only if you're using mutual TLS though right?  You wouldn't normally necessarily have to set a certfile and keyfile I don't think.

Reply:
And can you call out that this should only be called in hooks where the tls params are passed as arguments?  Which I think is only `get_outbound_tls_parameters` and `get_inbound_tls_parameters`

Reply:
Sure thing. I'll work on addressing this!